Jenkinsfile

```diff
@@ -32,7 +32,11 @@ void mvn(def args) {
     def mvnHome = tool name: 'mvn'
 
     /* Set JAVA_HOME, and special PATH variables. */
-    List javaEnv = ["PATH+JDK=${jdktool}/bin", "JAVA_HOME=${jdktool}"]
+    List javaEnv = [
+      "PATH+JDK=${jdktool}/bin", "JAVA_HOME=${jdktool}",
+      /* Additional variables needed by tests. */
+      'GIT_COMMITTER_EMAIL=me@hatescake.com','GIT_COMMITTER_NAME=Hates','GIT_AUTHOR_NAME=Cake','GIT_AUTHOR_EMAIL=hates@cake.com', 'LOGNAME=hatescake'
```

Comment:
The tests now pass on my OSX (dev and test) machine but fail without the env vars on my ubuntu 16 (test-only).  The ubuntu is purely a jenkins agent and has minimal configuration added.  

My current theory is that on a system without git user.name and user.email configured, a number of tests fail.  Once you add those settings (as I've done on my dev machine) the tests pass without the env vars.

It is better if the tests can run and pass  on clean(er) configurations - I've updated the comments to indicate that.